q2_amr/types/_format.py

```diff
@@ -431,6 +460,13 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
     def gene_path_maker(self, sample_id):
         return "%s/gene_mapping_data.txt" % sample_id
 
+    def sample_dict(self):
+        sample_dict = {}
+        for sample in self.path.iterdir():
+            file = list(os.path.join(sample, "gene_mapping_data.txt"))
```

Comment:
Let's not introduce variables where not necessary - you can also just assign the value directly below.